sdk/core/System.ClientModel/src/Options/ClientPipelineOptions.cs

```diff
@@ -103,22 +140,134 @@ public TimeSpan? NetworkTimeout
     }
 
     /// <summary>
-    /// The options to be used to create a <see cref="ClientLoggingPolicy"/>
-    /// if a custom <see cref="LoggingPolicy"/> is not provided. If
-    /// <see cref="LoggingPolicy"/> is not <c>null</c>, this property will
-    /// not be used.
+    /// Gets or sets the implementation of <see cref="ILoggerFactory"/> to use to
+    /// create <see cref="ILogger"/> instances for logging.
+    /// </summary>
+    /// <remarks>If an ILoggerFactory is not provided, logs will be written to Event Source
+    /// instead. If an ILoggerFactory is provided, logs will be written to ILogger only and not
+    /// Event Source.</remarks>
+    public ILoggerFactory? LoggerFactory
+    {
+        get => _loggerFactory;
+        set
+        {
+            AssertNotFrozen();
+
+            _loggerFactory = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets a list of header names that are not redacted during logging.
+    /// </summary>
+    /// <value>Defaults to a list of common header names that do not
+    /// typically hold sensitive information.</value>
+    public IList<string> AllowedHeaderNames
+    {
+        get => _allowedHeaderNames;
+    }
+
+    /// <summary>
+    /// Gets or sets a list of query parameter names that are not redacted during logging.
+    /// </summary>
+    /// <value>Defaults to a list of common query parameters that do not
+    /// typically hold sensitive information.</value>
+    public IList<string> AllowedQueryParameters
+    {
+        get => _allowedQueryParameters;
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if request and response content should be logged.
+    /// </summary>
+    /// <value>Defaults to <c>false</c>.</value>
+    public bool? EnableMessageContentLogging
+    {
+        get => _enableMessageContentLogging;
+        set
+        {
+            AssertNotFrozen();
+
+            _enableMessageContentLogging = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating if request and response content should be logged.
     /// </summary>
-    public LoggingOptions LoggingOptions
+    /// <value>Defaults to <c>false</c>.</value>
+    public bool? EnableMessageLogging
     {
-        get => _loggingOptions;
+        get => _enableMessageLogging;
         set
         {
             AssertNotFrozen();
 
-            _loggingOptions = value;
+            _enableMessageLogging = value;
         }
     }
 
+    /// <summary>
+    /// Gets or sets value indicating if request and response content should be logged.
+    /// </summary>
+    /// <value>Defaults to <c>false</c>.</value>
+    public bool? EnableLogging
+    {
+        get => _enableLogging;
+        set
+        {
+            AssertNotFrozen();
+
+            _enableLogging = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets value indicating maximum size of content to log in bytes.
+    /// </summary>
+    public int? MessageContentSizeLimit
```

Comment:
If this is on pipeline options instead of logging options, how will users know that it is the limit for the size that is logged and not the size of message content they can send with this pipeline?

Reply:
This is following the proposed naming for the API, but since the API was initially proposing a separate options type, this property will likely need a different name. I'll change it back to something more similar to what I had before.

Reply:
Let's get aligned on how we'll group the options first.